File: other/training/chat_servers_outgoing_traffic/answer/Myon.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    int member = 0;
    int ret = 0;
    string s;
    while (getline(cin, s))
    {
        if (s[0] == '+') member++;
        else if (s[0] == '-') member--;
        else
        {
            ret += member * (s.length() - s.find(":") - 1);
        }
        
    }
    cout << ret << endl;
    return 0;
}

File: other/training/chat_servers_outgoing_traffic/answer/Myon.php
<?php
function input()
{
    return rtrim(fgets(STDIN));
}

$member = 0;
$ret = 0;
while (true)
{
    $s = input() ?:"";
    if (strlen($s) == 0) break;
    $st = explode(':', $s);
    if (count($st) == 1)
    {
        if ($s[0] == '+') $member++;
        else $member--;
    }
    else
    {
        $ret += $member * strlen($st[1]);
    }
}
echo "$ret\n";
?>
